tests/Unit/Framework/MockRuntimeSystem.hpp

```diff
@@ -176,25 +176,23 @@ class MockRuntimeSystem {
     // passed the mock node and mock core associated with that cache.
     // The actual chares held in each cache are emplaced by the user explicitly
     // on a chosen node and core, via `emplace_component`.
-    for (const auto& node_and_core : mock_nodes_and_local_cores_) {
-      const size_t global_core =
-          mock_global_cores_.at(node_and_core.first).at(node_and_core.second);
+    for (const auto& [node,local_core] : mock_nodes_and_local_cores_) {
+      const size_t global_core = mock_global_cores_.at(node).at(local_core);
       mutable_caches_.at(global_core) =
                 std::make_unique<Parallel::MutableGlobalCache<Metavariables>>(
                     serialize_and_deserialize(mutable_cache_contents));
       caches_.at(global_core) = std::make_unique<GlobalCache>(
           serialize_and_deserialize(cache_contents),
           mutable_caches_.at(global_core).get());
       tmpl::for_each<typename Metavariables::component_list>(
-          [this, &global_core, &node_and_core](auto component) {
+          [this, &global_core, &node, &local_core](auto component) {
```

Comment:
I don't understand why this works with all compilers. Structures bindings are somehow different to variables and can't be captured.